wicket/src/ui/main.rs

```diff
@@ -0,0 +1,292 @@
+                        Some(Action::Redraw)
+                    } else {
+                        self.current_pane()
+                            .on(state, Event::Term(TermEvent::Key(e)))
+                    }
+                }
+                _ => {
+                    let event = Event::Term(TermEvent::Key(e));
+                    if self.sidebar.selected {
+                        self.sidebar.on(state, event)
+                    } else {
+                        self.current_pane()
+                            .on(state, Event::Term(TermEvent::Key(e)))
+                    }
+                }
+            },
+            e => {
+                if self.sidebar.selected {
+                    self.sidebar.on(state, e)
+                } else {
+                    self.current_pane().on(state, e)
+                }
+            }
+        }
+    }
+
+    fn current_pane(&mut self) -> &mut Box<dyn Pane> {
+        self.panes.get_mut(self.sidebar.selected()).unwrap()
+    }
+
+    fn draw_pane(
+        &mut self,
+        state: &State,
+        frame: &mut Frame<'_>,
+        tabs_rect: Rect,
+        pane_rect: Rect,
+    ) {
+        let (border_style, active, highlight_style) = if self.sidebar.selected {
+            (
+                style::deselected(),
+                false,
+                style::deselected().add_modifier(Modifier::BOLD),
+            )
+        } else {
+            (style::selected_line(), true, style::selected())
+        };
+
+        let pane = self.current_pane();
+
+        // Draw the Top bar (tabs)
+        let titles =
+            pane.tab_titles().iter().cloned().map(Spans::from).collect();
+        let tabs = Tabs::new(titles)
+            .highlight_style(highlight_style)
+            .divider(Span::styled("|", style::divider()))
+            .block(
+                Block::default()
+                    .borders(Borders::ALL)
+                    .border_type(BorderType::Rounded),
+            )
+            .style(border_style)
+            .select(pane.selected_tab());
+        frame.render_widget(tabs, tabs_rect);
+
+        // Draw the pane
+        pane.draw(state, frame, pane_rect, active)
+    }
+
+    fn draw_statusbar(
+        &mut self,
+        state: &State,
+        frame: &mut Frame<'_>,
+        rect: Rect,
+    ) {
+        let wicketd_spans =
+            state.service_status.wicketd_liveness.compute().to_spans();
+        let mgs_spans = state.service_status.mgs_liveness.compute().to_spans();
+        let mut spans = vec![Span::styled("WICKETD: ", style::service())];
+        spans.extend_from_slice(&wicketd_spans);
+        spans.push(Span::styled(" | ", style::divider()));
+        spans.push(Span::styled("MGS: ", style::service()));
+        spans.extend_from_slice(&mgs_spans);
+        let main = Paragraph::new(Spans::from(spans));
+        frame.render_widget(main, rect);
+
+        let test = Paragraph::new(Spans::from(vec![
```

Comment:
I'm not sure this makes sense for wicket during mupdate. Maybe we can put the currently operated version of the TUF repo there?

Reply:
Just double-checking definitions - by "currently operated version of the TUF repo" do you mean "system update version contained in the TUF repo"? I think the repo itself doesn't have a version.

Reply:
yep, that's what I meant.

Reply:
I'm going to do this in the follow up with update changes.